src/main/java/com/iexec/worker/pubsub/SubscriptionService.java

```diff
@@ -107,15 +114,54 @@ synchronized void reSubscribeToTopics() {
         Set<String> chainTaskIds = this.chainTaskIdToSubscription.keySet();
         if (chainTaskIds.isEmpty()) {
             log.info("No topic to resubscribe to");
-            return;
+        } else {
+            log.info("ReSubscribing to topics [chainTaskIds: {}]", chainTaskIds);
+            chainTaskIds.forEach(chainTaskId -> {
+                this.chainTaskIdToSubscription.remove(chainTaskId);
+                subscribeToTopic(chainTaskId);
+            });
+
+            log.info("ReSubscribed to topics [chainTaskIds: {}]", chainTaskIds);
         }
 
-        log.info("ReSubscribing to topics [chainTaskIds: {}]", chainTaskIds);
-        chainTaskIds.forEach(chainTaskId -> {
-            this.chainTaskIdToSubscription.remove(chainTaskId);
-            subscribeToTopic(chainTaskId);    
-        });
-        log.info("ReSubscribed to topics [chainTaskIds: {}]", chainTaskIds);
+        if (sessionReady) {
+            log.warn("STOMP session was already up before receiving this event");
+        }
+
+        sessionReady = true;
+        sessionLock.lock();
+        try {
+            sessionReadyCondition.signalAll();
+        } finally {
+            sessionLock.unlock();
+        }
+    }
+
+    @EventListener(SessionLostEvent.class)
+    synchronized void sessionLost() {
+        log.warn("STOMP session is down, blocking updates until it is restored.");
+        if (!sessionReady) {
+            log.warn("STOMP session was already down before receiving this event");
+        }
+        sessionReady = false;
+    }
+
+    /**
+     * Wait for the session to be ready.
+     * Useful to prevent actions to execute while the STOMP session is disconnected.
+     *
+     * @throws InterruptedException if the current thread is interrupted
+     *         while waiting.
+     */
+    public void waitForSessionReady() throws InterruptedException {
+        while (!sessionReady) {
+            sessionLock.lock();
+            try {
+                sessionReadyCondition.await();
+            } finally {
+                sessionLock.unlock();
+            }
+        }
```

Comment:
I don't know if the implementation couldn't be simplified.
https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/concurrent/locks/Condition.html
https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#wait(long,int)
```suggestion
        synchronized(obj) {
            while (!sessionReady) {
                obj.wait();
            }
        }
```

The call to `signalAll` would be replaced with a call to `notifyAll`.
This could be done in another ticket to evaluate if it works as a lighter implementation.

Reply:
I would avoid synchronizing on an object there. If you do so, while you'll be in the `while` block, you won't be able to enter another `synchronized` block. Which means you won't be able to call `signalAll`. Do you see my point?

Reply:
As said, it is an open question for a next iteration. Maybe it is the proposed synchronization which is bad.
If it was so unusable, it would not exist in the language.
See https://www.baeldung.com/java-wait-notify for further thinking on this subject.